diff --git a/dwm.c b/dwm.c
index 582c9c9..f8bb823 100644
--- a/dwm.c
+++ b/dwm.c
@@ -707,11 +707,11 @@ drawbar(Monitor *m)
 	Client *c;
 
 	/* draw status first so it can be overdrawn by tags later */
-	if (m == selmon) { /* status is only drawn on selected monitor */
-		drw_setscheme(drw, scheme[SchemeNorm]);
-		tw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
-		drw_text(drw, m->ww - tw - 2 * sp, 0, tw, bh, 0, stext, 0);
-	}
+	/* removed conditional here to only draw on selected monitor, now draws on all */
+	drw_setscheme(drw, scheme[SchemeNorm]);
+	tw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
+	drw_text(drw, m->ww - tw - 2 * sp, 0, tw, bh, 0, stext, 0);
+
 
 	for (c = m->clients; c; c = c->next) {
 		occ |= c->tags;
